### PR TITLE
Add Markdown link check to GitHub Actions

### LINKED ISSUE #64: Add a link checker to this project
There are various tools that will check links in GitHub Actions, such as [lychee link checker action](https://github.com/marketplace/actions/lychee-link-checker-action). Given the number of links in the README for this project, it would be good to have something like that installed that checks maybe every month and on PRs so we hear about it if links break.

### PR DESCRIPTION
**Pull request by NicMcPhee**
Adds a `link-check.yaml` GitHub Actions workflow file that uses the `gaurav-nelson/github-action-markdown-link-check` action to check our links.

I didn't turn on the `modified-files-only` feature, but that may be something we want to do down the road.

I also didn't add scheduled runs, but it might be worth having this run every month or so rather than just have it happen on push events.

Closes #64

### CODE REVIEW

**Comment by NicMcPhee** [comment]
It looks like the badge links fail the link check, so we'll need to add some config telling it to ignore those links.

File: .github/mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "ignorePatterns": [
+    {
+      "pattern": "^\.\./\.\./actions/workflows"
```
Comment (NicMcPhee) [inline .github/mlc_config.json:4]: This ignores the badge URLs, which seem to not check out (probably because of the relative paths?).

**Review by helloworld12321 — APPROVED** [review]
LGTM!

Re: cron job, I could go either way on that. I'd lean towards not having a cron job, just for simplicity's sake. 🙂

Thanks much!

**Comment by floogulinc** [comment]
Since this wouldn't be a requires check you could even filter when it runs to only when a markdown file has been changed.

**Pull request merged**